python/ray/autoscaler/_private/_azure/node_provider.py

```diff
@@ -163,6 +163,55 @@ def _extract_metadata(self, vm):
 
         return metadata
 
+    def _get_zones_for_vm_size(self, vm_size, location):
+        """Get usable availability zones for a given VM size in a specific location."""
+        try:
+            # Note: Azure ResourceSKUs API filters don't work reliably(?), so we query all SKUs
+            # and filter in code. Each SKU object represents one location for the VM size.
+            skus = self.compute_client.resource_skus.list()
```

Comment:
Is this expensive?

Reply:
I don't think it is expensive. It is a lot less expensive than making multiple calls to the azure SDK